CMakePresets.json

```diff
@@ -0,0 +1,60 @@
+{
+    "version": 3,
+    "configurePresets": [
+        {
+            "name": "default",
+            "hidden": true,
+            "generator": "Ninja",
+            "binaryDir": "${sourceDir}/build/${presetName}",
```

Comment:
I'm wondering if you did some testing prior to adding this line, @reecewayt, and maybe that's why you have config files directly in the root of your `build` directory. Would you want to try deleting that directory on your machine and see if you end up with just a `build` directory with subdirectories for each preset like I have? Might help us in troubleshooting setup and configuration across everyone's machines to know.